supervisor/shared/filesystem.c

```diff
@@ -69,6 +69,18 @@ static void make_empty_file(FATFS *fatfs, const char *path) {
     f_close(&fp);
 }
 
+
+static void make_sample_code_file(FATFS *fatfs) {
+    FIL fs;
+    UINT *char_written = 0;
+    const byte buffer[] = "print('Hello World!')";
+
+    //Create or modify existing code.py file 
+    f_open(fatfs, &fs, "/code.py", FA_WRITE | FA_CREATE_ALWAYS);
+    f_write(&fs, buffer, sizeof(buffer), char_written);
```

Comment:
Passing the NULL pointer value `char_written` here is not correct, though the problem may go undiagnosed.  Unlike other languages, C does not have "out" or "reference" parameters.  Instead, the programmer manually emulates them by passing the *address of* some suitable value using the `&` operator.

Recommendation: The declaration should be `UINT char_written = 0;`, and the address of it passed to `f_write`:  `f_write(..., &char_written)`.

Reply:
Agreed!